Can't build on VS2015 when paths have spaces , missing 'microsoft-scripting-chakra-instrumentationevents.h'

following the instructions in the doc and I can't build. I also can't seem to find this header file in the repo. 

Severity	Code	Description	Project	File	Line	Suppression State
Error		0x2 trying to open file <Studio>.	CoreManifests	C:\Users\bill\Documents\Visual Studio 2015\Projects\ChakraCore-master\ChakraCore-master\manifests\mc	1	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.WasmReader	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.Math	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.ByteCode	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.Language	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Common.Core	C:\Users\bill\Documents\Visual Studio 2015\Projects\ChakraCore-master\ChakraCore-master\lib\Common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.Debug	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.Library	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.Types	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.Base	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Common.Common	C:\Users\bill\Documents\Visual Studio 2015\Projects\ChakraCore-master\ChakraCore-master\lib\Common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Common.Memory	C:\Users\bill\Documents\Visual Studio 2015\Projects\ChakraCore-master\ChakraCore-master\lib\Common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Parser	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Backend	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Jsrt	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Jsrt.Core	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Jsrt.Core	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.PlatformAgnostic	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.PlatformAgnostic	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.PlatformAgnostic	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.JITServer	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.PlatformAgnostic	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.PlatformAgnostic	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.PlatformAgnostic	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	Chakra.Runtime.PlatformAgnostic	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	GCStress	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	ChakraCore	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	GCStress	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	ChakraCore	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	GCStress	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	ChakraCore	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	C1083	Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory	GCStress	c:\users\bill\documents\visual studio 2015\projects\chakracore-master\chakracore-master\lib\common\Core\EtwTraceCore.h	68	
Error	LNK1181	cannot open input file 'C:\Users\bill\Documents\Visual Studio 2015\Projects\ChakraCore-master\ChakraCore-master\Build\VcBuild\bin\x86_test\ChakraCore.lib'	NativeTests	C:\Users\bill\Documents\Visual Studio 2015\Projects\ChakraCore-master\ChakraCore-

Unless I'm missing something, this looks like an issue related to how the build system handles paths with spaces. I can build ChakraCore just fine, either through Visual Studio or msbuild, from paths without spaces.

When the path has a space in it, on the other hand, I get the same errors @JohnathanNorman is:

```
mc : error : 0x3 trying to open file <Copy\Build\VcBuild\obj\x86_release\CoreManifests\>.
c:\chakracore - copy\lib\common\Core/EtwTraceCore.h(68): fatal error C1083: Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory
```

Build logs: [Visual Studio](https://gist.github.com/WickedA/15d04dc606599b42bba9dcb40b3ce86c), [MSBuild](https://gist.github.com/WickedA/3d07d4a8d91c18ec732cc1db98c57ed3).

Yes that is the issue
 I changed directories and it works now. 

From: Andy C.
Sent: Tuesday, January 31, 2017 12:26 AM
To: Microsoft/ChakraCore
Cc: JohnathanNorman; Mention
Subject: Re: [Microsoft/ChakraCore] Can't build on VS2015 , missing'microsoft-scripting-chakra-instrumentationevents.h' (#2422)

Unless I'm missing something, this looks like an issue related to how the build system handles paths with spaces. I can build ChakraCore just fine, either through Visual Studio or msbuild, from paths without spaces.
When the path has a space in it, on the other hand, I get the same errors @JohnathanNorman is:
mc : error : 0x3 trying to open file <Copy\Build\VcBuild\obj\x86_release\CoreManifests\>.
c:\chakracore - copy\lib\common\Core/EtwTraceCore.h(68): fatal error C1083: Cannot open include file: 'microsoft-scripting-chakra-instrumentationevents.h': No such file or directory
Build logs: Visual Studio, MSBuild.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

--- a/lib/JITIDL/Chakra.JITIDL.vcxproj
+++ b/lib/JITIDL/Chakra.JITIDL.vcxproj
@@ -31,7 +31,7 @@
   <ItemGroup>
     <Midl Include="$(MsBuildThisFileDirectory)ChakraJIT.idl">
       <OutputDirectory>$(IntDir)</OutputDirectory>
-      <AdditionalOptions>%(AdditionalOptions) /acf $(MsBuildThisFileDirectory)ChakraJIT.acf /prefix client "Client" server "Server"</AdditionalOptions>
+      <AdditionalOptions>%(AdditionalOptions) /acf "$(MsBuildThisFileDirectory)ChakraJIT.acf" /prefix client "Client" server "Server"</AdditionalOptions>
     </Midl>
   </ItemGroup>
   <ItemGroup>

--- a/manifests/CoreManifests.vcxproj
+++ b/manifests/CoreManifests.vcxproj
@@ -16,7 +16,7 @@
   <ItemGroup>
     <CustomBuild Include="$(MSBuildThisFileDirectory)Microsoft-Scripting-Chakra-Instrumentation.man">
       <FileType>Document</FileType>
-      <Command>mc.exe /h $(IntDir) /W "$(EventManifestXmlPath)\winmeta.xml" /w "$(EventManifestXmlPath)\eventman.xsd" /r $(IntDir) /z %(Filename)Events -um /v %(FullPath)</Command>
+      <Command>mc.exe /h "$(IntDir)." /W "$(EventManifestXmlPath)\winmeta.xml" /w "$(EventManifestXmlPath)\eventman.xsd" /r "$(IntDir)." /z %(Filename)Events -um /v "%(FullPath)"</Command>
       <Outputs>$(IntDir)\%(Filename)Events.h;$(IntDir)\%(Filename)Events.rc;$(IntDir)\%(Filename)Events_MSG00001.bin;$(IntDir)\%(Filename)EventsTEMP.bin</Outputs>
     </CustomBuild>
   </ItemGroup>

